submissions/NastjonkaK/OOP_exercise/index.js

```diff
@@ -3,82 +3,95 @@ class Creature {
     this.name = name,
     this.saying = saying
   }
+
+  introduction() {
+     return `\Hello! I am a ${this.type}, ${this.gender}. My name is ${this.name}.`
+    }
+
+  say(){
+    return `\And I want to say: "${this.saying}".`
+  }
 }
 
-class Animals extends Creature {
-  constructor(name, gender, saying) {
+class Animal extends Creature {
+  constructor(name, gender, paws, saying) {
     super(name, saying);
     this.gender = gender,
-    this.paws = 4
+    this.paws = paws
   }
 
   greeting() {
-    return `\ Hello! I am a ${this.type}, ${this.gender}. My name is ${this.name}. I have ${this.paws} paws and a tail. And I like to say: "${this.saying}".`;
+    return `\ ${this.introduction()} I have ${this.paws} paws and a tail. ${this.say()}`;
   }
 }
 
 class Human extends Creature {
-  constructor(name, saying) {
-    super(name, saying)
+  constructor(name, legs, hands, saying) {
+    super(name, saying);
     this.type = "human",
-    this.legs = 2,
-    this.hands = 2
+    this.legs = legs,
+    this.hands = hands
   }
 
   greeting() {
-    return `\ Hello! I am a ${this.type}, ${this.gender}. My name is ${this.name}. I have ${this.legs} legs and ${this.hands} hands. And I want to say: "${this.saying}".`;
+    return `\ ${this.introduction()}  I have ${this.legs} legs and ${this.hands} hands. ${this.say()}`;
   }
 }
 
+const MyMixin = (superclass) => class extends superclass {
+    constructor(name, gender,legs, paws) {
+      super(name, gender, legs, paws);
+      this.saying = "meow-meow"
+    }
+};
+
 class Man extends Human {
-  constructor(name, saying) {
-    super(name, saying);
+  constructor(name, legs, hands, saying) {
+    super(name, legs, hands, saying);
     this.type = 'man',
     this.gender = 'male'
   }
 }
 
 class Woman extends Human {
-  constructor(name, saying) {
-    super(name, saying);
+  constructor(name, legs, hands, saying) {
+    super(name, legs, hands, saying);
     this.type = 'woman',
     this.gender = 'female'
   }
 }
 
 
-class Dog extends Animals {
-  constructor(name, gender) {
-    super(name, gender);
+class Dog extends Animal {
+  constructor(name, gender, paws) {
+    super(name, gender, paws);
     this.type = "dog",
     this.saying = "woof-woof"
   }
 }
 
-class Cat extends Animals {
-  constructor(name, gender) {
-    super(name, gender);
-    this.type = "cat",
-    this.saying = "meow-meow"
+class Cat extends MyMixin(Animal) {
+  constructor(name, gender, paws) {
+    super(name, gender, paws);
+    this.type = "cat"
   }
 }
 
-class HumanCat extends Woman {
-  constructor(name, saying) {
-    super(name, saying);
-    this.type = "human-cat",
-    this.gender = "female"
+class HumanCat extends MyMixin(Woman) {
```

Comment:
`whenever you change cat's saying cat-woman's saying should change accordingly,`
MyMixin saying is not cat's, btw idea is good.
Lets try one more time :smiley_cat: